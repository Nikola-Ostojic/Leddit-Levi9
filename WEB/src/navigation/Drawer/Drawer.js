import React from "react";
import MaterialDrawer from "@material-ui/core/Drawer";
import List from "@material-ui/core/List";
import Divider from "@material-ui/core/Divider";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import HomeIcon from "@material-ui/icons/Home";
import StarIcon from "@material-ui/icons/Star";
import { makeStyles } from "@material-ui/core/styles";
import { withRouter } from "react-router-dom";

const useStyles = makeStyles((theme) => ({
  drawer: {
    width: 240,
    flexShrink: 0
  },
  drawerPaper: {
    width: 240,
    backgroundColor: theme.palette.primary.main,
    color: theme.palette.common.white
  },
  toolbar: theme.mixins.toolbar
}));

export const Drawer = ({ drawerOpen, setDrawerOpen, history: { push } }) => {
  const classes = useStyles();

  return (
    <MaterialDrawer
      className={classes.drawer}
      variant="temporary"
      classes={{
        paper: classes.drawerPaper
      }}
      open={drawerOpen}
      onClose={() => setDrawerOpen(false)}
    >
      <div className={classes.toolbar}></div>
      <div role="presentation" onClick={() => setDrawerOpen(false)}>
        <List>
          <ListItem button onClick={() => push("")}>
            <ListItemIcon>
              <HomeIcon htmlColor="white" />
            </ListItemIcon>
            <ListItemText primary="Home" />
          </ListItem>
        </List>
        <Divider />
        <List>
          <ListItem button onClick={() => push("/movies")}>
            <ListItemIcon>
              <StarIcon htmlColor="white" />
            </ListItemIcon>
            <ListItemText primary="Movies" />
          </ListItem>
        </List>
      </div>
    </MaterialDrawer>
  );
};

export default withRouter(Drawer);
