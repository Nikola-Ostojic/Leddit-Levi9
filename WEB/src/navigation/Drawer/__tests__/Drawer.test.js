import React from "react";
import { shallow } from "enzyme";
import MaterialDrawer from "@material-ui/core/Drawer";
import ListItem from "@material-ui/core/ListItem";

import { Drawer } from "../Drawer";

describe("Drawer component tests", () => {
  const mockedProps = {
    drawerOpen: true,
    setDrawerOpen: jest.fn(),
    history: { push: jest.fn() }
  };
  it("Should match snapshot", () => {
    const wrapper = shallow(<Drawer {...mockedProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("Should call setDrawerOpen when material drawer is pressed", () => {
    const wrapper = shallow(<Drawer {...mockedProps} />);

    wrapper.find(MaterialDrawer).simulate("close");
    expect(mockedProps.setDrawerOpen).toHaveBeenCalledWith(
      !mockedProps.drawerOpen
    );
  });

  it("Should call push to home on first list item click", () => {
    const wrapper = shallow(<Drawer {...mockedProps} />);

    wrapper.find(ListItem).at(0).simulate("click");
    expect(mockedProps.history.push).toHaveBeenCalledWith("");
  });

  it("Should call push to movies on second list item click", () => {
    const wrapper = shallow(<Drawer {...mockedProps} />);

    wrapper.find(ListItem).at(1).simulate("click");
    expect(mockedProps.history.push).toHaveBeenCalledWith("/movies");
  });

  it("Should call setDrawerOpen with false when the presentation div is clicked", () => {
    const wrapper = shallow(<Drawer {...mockedProps} />);

    wrapper.find('[role="presentation"]').simulate("click");
    expect(mockedProps.setDrawerOpen).toHaveBeenCalledWith(false);
  });
});
