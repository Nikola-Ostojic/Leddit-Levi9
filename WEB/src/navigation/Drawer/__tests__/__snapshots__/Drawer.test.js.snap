// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer component tests Should match snapshot 1`] = `
<WithStyles(ForwardRef(Drawer))
  className="makeStyles-drawer-1"
  classes={
    Object {
      "paper": "makeStyles-drawerPaper-2",
    }
  }
  onClose={[Function]}
  open={true}
  variant="temporary"
>
  <div
    className="makeStyles-toolbar-3"
  />
  <div
    onClick={[Function]}
    role="presentation"
  >
    <WithStyles(ForwardRef(List))>
      <WithStyles(ForwardRef(ListItem))
        button={true}
        onClick={[Function]}
      >
        <WithStyles(ForwardRef(ListItemIcon))>
          <HomeIcon
            htmlColor="white"
          />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Home"
        />
      </WithStyles(ForwardRef(ListItem))>
    </WithStyles(ForwardRef(List))>
    <WithStyles(ForwardRef(Divider)) />
    <WithStyles(ForwardRef(List))>
      <WithStyles(ForwardRef(ListItem))
        button={true}
        onClick={[Function]}
      >
        <WithStyles(ForwardRef(ListItemIcon))>
          <StarIcon
            htmlColor="white"
          />
        </WithStyles(ForwardRef(ListItemIcon))>
        <WithStyles(ForwardRef(ListItemText))
          primary="Movies"
        />
      </WithStyles(ForwardRef(ListItem))>
    </WithStyles(ForwardRef(List))>
  </div>
</WithStyles(ForwardRef(Drawer))>
`;
